src/main/kotlin/org/elm/lang/core/types/Ty.kt

```diff
@@ -224,34 +230,42 @@ object TyInProgressBinding : Ty() {
 data class AliasInfo(val module: String, val name: String, val parameters: List<Ty>)
 
 
-/** Create a lazy sequence of all [TyVar]s referenced within this ty. */
-fun Ty.allVars(includeAlias: Boolean = false): Sequence<TyVar> = traverse(includeAlias).filterIsInstance<TyVar>()
+/** Return a list of all [TyVar]s in this ty, including itself */
+fun Ty.allVars(): List<TyVar> = mutableListOf<TyVar>().also { allVars(it) }
 
-fun Ty.traverse(includeAlias: Boolean = false): Sequence<Ty> = sequence {
-    yield(this@traverse)
-    when (this@traverse) {
-        is TyTuple -> types.forEach { yieldAll(it.traverse(includeAlias)) }
+private fun Ty.allVars(result: MutableList<TyVar>) {
```

Comment:
This code was very hot, so I replaced the coroutine-based `traverse` with hand-coded recursive descent for each use-case. Surprisingly, the coroutines were only ~25% slower, so maybe the extra code isn't worth it.